src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1415,10 +1415,194 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
         }
     }
 
+    bool       isScalar = false;
+    genTreeOps oper     = node->HWOperGet(&isScalar);
+
+    switch (oper)
+    {
+        case GT_AND:
+        case GT_OR:
+        case GT_XOR:
+        {
+            if (!comp->canUseEvexEncoding())
+            {
+                break;
+            }
+
+            GenTree* op1 = node->Op(1);
+            GenTree* op2 = node->Op(2);
+
+            LIR::Use use;
+            if (BlockRange().TryGetUse(node, &use))
+            {
+                // search for structure like:
+                /*
+                        /- A
+                        +- B
+                    t1 = binary logical op1
+
+                        /- C
+                        +- t1
+                    t2 = binary logical op2
+                */
+                GenTree* second = use.User();
+                if (!second->OperIs(GT_HWINTRINSIC))
+                {
+                    break;
+                }
+
+                bool       nestedIsScalar = false;
+                genTreeOps nestedOper     = second->AsHWIntrinsic()->HWOperGet(&isScalar);
+
+                if (nestedOper == GT_NONE)
```

Comment:
don't think we need this condition. It will be taken care by the condition on line 1468 below.

Reply:
The one further down is temporary and will be removed so we can handle other cases such as not, and_not, and other other future binary operations